Generator.Equals/Models/AttributeMetadata.cs

```diff
@@ -0,0 +1,45 @@
+using System;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+
+namespace Generator.Equals.Models;
+
+internal sealed record AttributeMetadata(string FullName, string Namespace, string LongName, string ShortName)
```

Comment:
Guessing this doesn't handle nested types, does it need to?

Reply:
What do you mean?

Reply:
I mean a type nested inside another type.

Reply:
> What do you mean? I think...not?

If I was a casual user, I think I might be upset to use this library and find that the following code was unsupported:
```cs
public partial class Outer
{
  [GenerateEquality]
  private partial class Inner
  {
  }
}
```

Reply:
Yea this works. It may lead to errors if there are attributes in our code with the same name. (like ``GenerateEquality``)